### PR TITLE
Mostly auto-generate types

### PR DESCRIPTION
**Pull request by kainino0x**
- [x] `cts` still typechecks

Should be reviewed per-commit (many are no-ops and can be ignored or skimmed)

### CODE REVIEW

**Comment by kainino0x** [comment]
@austinEng PTAL!

@darionco FYI

**Comment by darionco** [comment]
@kainino0x I have addressed all the issues you reported on [bikeshed-to-ts](https://github.com/darionco/bikeshed-to-ts) the only sticking point is `Iterable` vs `Array` since they are not a 1 to 1.

I released the changes in version `1.2.0` I tested it with the latest version of the spec and seemed to produce good results. The dif will be hard once more because it now sorts the declarations.

Let me know how it goes or if you want me to submit a PR with the new version.

**Comment by kainino0x** [comment]
Amazing, thank you @darionco!! I will find time to try it out this coming week.

File: README.md [hunk 1]
```diff
@@ -55,7 +55,47 @@ In `webpack.config.js` add:
...
+- `| null` added for nullable types.
+- `__brand` removed for partial interfaces.
+- `__brand` removed for WebIDL `dictionary`s.
+- `var`s added for WebIDL `interface`s.
+- `Array` changed to `Iterable` for WebIDL `sequence`s.
+- `any` changed to `object` for WebIDL `object`.
+- `interface` changed to `var` for WebIDL `interface`s with (only) `const` members.
+    (TODO: this isn't quite the right translation either.)
+- `new ()` moved from `interface` to `var` for constructible WebIDL `interface`s.
+- `| SharedArrayBuffer` added for `[AllowShared]` types.
```
Comment (austinEng) [inline README.md:84]: should this only be for [AllowShared] on an `ArrayBuffer` type?

ex.) `foo(arr: [AllowShared] Float32Array): void` probably shouldn't accept `SharedArrayBuffer`

**Review by austinEng — APPROVED** [review]
LGTM

**Pull request merged**